### DISCUSSION TITLE
Cast not printing hex values

### DISCUSSION THREAD

**Discussion by 0xmulch** [Cyfrin/foundry-full-course-cu#1341] [Q&A] [answered]
Hi all, when running foundry-simple-storage locally w/ anvil, I run the following commands and I am unable to retrieve my hex values it seems. Windows 11- was hoping someone could help me with this:

```
~/workshop/foundry-simple-storage:> cast call <contractId> "addPerson(string, u
int256)" ben 777
0x

~/workshop/foundry-simple-storage:> cast call <contractId> "nameToFavoriteNumber(string)" ben
0x
```
Screenshot below of it working in Remix:
![Screenshot 2024-01-19 064359](https://github.com/Cyfrin/foundry-full-course-f23/assets/156262846/4ca1a1bb-22cb-4d32-b7c6-90d745cf59c9)

**Comment by 0xusmanf** [comment]
> ~/workshop/foundry-simple-storage:> cast call <contractId> "addPerson(string, u
> int256)" ben 777
> 0x

`cast call` can only read from a contract and here you are trying to write to the contract using `addPerson` function. To write to a contract you can use `cast send` command just like you use `cast call`, also make sure there are no spaces in the function definition.

> ~/workshop/foundry-simple-storage:> cast call <contractId> "nameToFavoriteNumber(string)" ben
> 0x

After you write successfully to the contract using `cast send`, this command should work fine.

**Reply by 0xmulch** [reply]
I will try these ASAP, thanks @usmanfarooq91

**Reply by 0xmulch** [reply]
@usmanfarooq91 still seem to be unable to print that friggen hex value

```
~/workshop/foundry-simple-storage:> cast send <cId> "addPerson(string)(uint256)" test 5 --private-key $PRIVATE_KEY

blockHash               0x573ab222ff31b014294e9211d9b4a3201c0bc3eec0de40472d86c90b77600906
blockNumber             3
contractAddress
cumulativeGasUsed       21520
effectiveGasPrice       3765938867
gasUsed                 21520
logs                    []
logsBloom               0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
root
status                  1
transactionHash         0xdb841ba8b4c4ba579eb270436736a5aa111a2ece16a13e31444625b60dc6b6eb
transactionIndex        0
type                    2
~/workshop/foundry-simple-storage:> cast call <cId> "nameToFavoriteNumber(string)" "test"
0x
```
Tried doing `call` w/ and w/o my 'test' being in quotes

**Reply by 0xmulch** [reply, marked as answer]
Got it! Because we're calling on a mapping, we need to encode the arguments first:

```
cast call 0x34A1D3fff3958843C43aD80F30b94c510645C316 "nameToFavoriteNumber(string)" 0x8bab8dd5000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000076272616e646f6e00000000000000000000000000000000000000000000000000
```
